homeassistant/components/enocean/cover.py

```diff
@@ -0,0 +1,267 @@
+"""Support for EnOcean roller shutters (EEP D2-05-00)."""
+from __future__ import annotations
+
+import asyncio
+import logging
+
+from enocean.protocol.packet import Packet  # pylint: disable=import-error
+from enocean.utils import combine_hex  # pylint: disable=import-error
+import voluptuous as vol  # pylint: disable=import-error
+
+from homeassistant.components.cover import (
+    ATTR_POSITION,
+    DEVICE_CLASS_BLIND,
+    DEVICE_CLASSES_SCHEMA,
+    PLATFORM_SCHEMA,
+    CoverEntity,
+    CoverEntityFeature,
+)
+from homeassistant.const import CONF_DEVICE_CLASS, CONF_ID, CONF_NAME
+from homeassistant.core import HomeAssistant
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.dispatcher import dispatcher_send
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
+
+from .const import SIGNAL_SEND_MESSAGE
+from .device import EnOceanEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = "EnOcean roller shutter"
+DEPENDENCIES = ["enocean"]
+
+CONF_SENDER_ID = "sender_id"
+WATCHDOG_TIMEOUT = "watchdog_timeout"
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_ID): vol.All(cv.ensure_list, [vol.Coerce(int)]),
+        vol.Required(CONF_SENDER_ID): vol.All(cv.ensure_list, [vol.Coerce(int)]),
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Optional(CONF_DEVICE_CLASS): DEVICE_CLASSES_SCHEMA,
+        vol.Optional(WATCHDOG_TIMEOUT, default=5): int,
```

Comment:
Do we need an option for the timeout, or is a sane default enough?

Reply:
Thank you very much for your review! In my view an option is required, as the timeout depends on the open/close time of the connected device.

Reply:
Is it expensive for the device if we send a telegram to it? Ie, is it not good to set a short timeout and send a telegram frequently to the device?

Reply:
Good idea - see commit [a1631a5](https://github.com/home-assistant/core/pull/75356/commits/a1631a5dcb2c579b544cab604dac5f9a39c174ac). 

Some more remarks to that commit:
- Due to the shorter interval, I also had to adjust the status detection (because now two subsequent received statuses might have the same position info even though the roller shutter is still moving). 
- Moreover, I also added a query limit, preventing the watchdog to run forever (i.e. if a device is unpowered or defective). 

Finally, I have also added a status query after (re)start of Home Assistant - see commit [80caa02](https://github.com/home-assistant/core/pull/75356/commits/80caa02c7b49b43c99ea16d853adc63aa8d6fd51).

In my view, all your comments / change request should be covered now - please have a look and let me know if there is anything that still needs adjustments.

Reply:
To also answer your question: no it is not expensive for the device (except, probably, some negligible additional energy consumption). However, in my experience EnOcean transmissions are a bit unreliable (probably due to collisions) and hence I try not to make too many / too frequent queries. I have tested the changes suggested in my commits in my current setup (ten roller shutters) - they seem to be working fine.